src/directives/formly-form.test.js

```diff
@@ -358,6 +358,25 @@ describe('formly-form', () => {
 
   });
 
+  describe('hideExpression', () => {
+    beforeEach(() => {
+      scope.model = {};
+      scope.fieldModel = {};
+
+      scope.fields = [
+        {template: input, key: 'foo', model: scope.fieldModel},
+        {template: input, key: 'bar', model: scope.fieldModel, hideExpression: () => !!scope.fieldModel.foo}
+      ];
+    });
+
+    it('should be updated when field model changes', () => {
```

Comment:
Fantastic! Could you add another test that checks the opposite case? Where the field's `hide` property is initialized to `true` and then is swapped to `false` via the `hideExpression`? I think that would be a valuable test case.